src/bot/exts/dragonfly/dragonfly.py

```diff
@@ -582,6 +583,35 @@ async def before_scan_loop(self: Self) -> None:
         """Wait until the bot is ready."""
         await self.bot.wait_until_ready()
 
+    @commands.has_role(Roles.vipyr_security)
+    @commands.hybrid_command()
+    async def queue(self: Self, ctx: commands.Context, name: str, version: str) -> None:
+        """Add a package to the Dragonfly scan queue."""
+        try:
+            await self.bot.dragonfly_services.queue_package(name=name, version=version)
+            await ctx.send(f"Successfully queued package `{name} v{version}`")
+        except aiohttp.ClientResponseError as http_error:
+            status_code = http_error.status
+
+            if status_code == HTTPStatus.NOT_FOUND:
+                await ctx.send(f"Package `{name} v{version}` was not found on PyPI")
+
+            if status_code == HTTPStatus.CONFLICT:
+                scan_results: list[Package] = await self.bot.dragonfly_services.get_scanned_packages(
```

Comment:
Is this typehint necessary?

Reply:
Also, it would be nice to avoid making two requests; that would mean the error from mainframe contains the existing package information. It doesn't seem difficult to implement, but I don't think it matters that much. Thoughts?

Reply:
yah, I'm not a fan of having to do the extra API call, but Mainframe doesn't provide enough information in the response for us to differentiate between a package already queued and package already scanned. It may be worth looking into in a separate PR but for now, I think this is fine as I don't suspect this command being used frequently enough to warrant that.